File: android/app/src/main/java/com/aerion/amrosa/domain/model/ReceivedRecipeSummary.kt
package com.aerion.amrosa.domain.model

/** Card-level data for a recipe shared directly to the current user. */
data class ReceivedRecipeSummary(
    val shareId: String,
    val title: String,
    /** Original recipe author (e.g. "Tanay" or "Imported"). */
    val authorDisplayName: String,
    /** Who sent this recipe to you. */
    val fromDisplayName: String,
    val sharedAt: Long,
    val prepTimeMinutes: Int?,
    val cookTimeMinutes: Int?,
    val tags: List<String>
)

/** Full recipe + sender name, returned by getReceivedRecipe(). */
data class ReceivedRecipeData(
    val recipe: Recipe,
    val fromDisplayName: String
)

File: ios/Amrosa/Amrosa/Data/DTOs/ReceivedRecipeSummary.swift
import Foundation

/// Card-level data for a recipe shared directly to the current user.
struct ReceivedRecipeSummary: Identifiable {
    let shareId: String
    let title: String
    /// Original recipe author (e.g. "Tanay" or "Imported").
    let authorDisplayName: String
    /// Who sent this recipe to you.
    let fromDisplayName: String
    let sharedAt: Date
    let prepTimeMinutes: Int?
    let cookTimeMinutes: Int?
    let tags: [String]

    var id: String { shareId }
}

/// Full recipe + sender name, returned by getReceivedRecipe().
struct ReceivedRecipeData {
    let recipe: SharedRecipe
    let fromDisplayName: String
}
